tests/topotests/bgp_remove_private_as/test_bgp_remove_private_as.py

```diff
@@ -0,0 +1,437 @@
+                    "100.64.0.2/32": "5555 1000 2000",
+                    "100.64.0.3/32": "5555 1000 2000",
+                    "100.64.0.4/32": "5555 1000 2000 2000 3000",
+                },
+            },
+            "remove-private-AS replace-AS": {
+                "r2": {
+                    "100.64.0.0/32": "65002 65002 65002 65002 65002",
+                    "100.64.0.1/32": "65002 65002 65002 65002 65002 65002 65002",
+                    "100.64.0.2/32": "65002 65001 4200000000 1000 4200000001 2000 4200000002",
+                    "100.64.0.3/32": "65002 65001 65003 4200000000 1000 4200000001 2000 4200000002 65003",
+                    "100.64.0.4/32": "65002 65001 1000 2000 2000 3000",
+                },
+                "r5": {
+                    "100.64.0.0/32": "5555 5555 5555 5555 5555",
+                    "100.64.0.1/32": "5555 5555 5555 5555 5555 5555 5555",
+                    "100.64.0.2/32": "5555 65001 4200000000 1000 4200000001 2000 4200000002",
+                    "100.64.0.3/32": "5555 65001 65003 4200000000 1000 4200000001 2000 4200000002 65003",
+                    "100.64.0.4/32": "5555 65001 1000 2000 2000 3000",
+                },
+            },
+            "remove-private-AS all replace-AS": {
+                "r2": {
+                    "100.64.0.0/32": "65002 65002 65002 65002 65002",
+                    "100.64.0.1/32": "65002 65002 65002 65002 65002 65002 65002",
+                    "100.64.0.2/32": "65002 65002 65002 1000 65002 2000 65002",
+                    "100.64.0.3/32": "65002 65002 65002 65002 1000 65002 2000 65002 65002",
+                    "100.64.0.4/32": "65002 65002 1000 2000 2000 3000",
+                },
+                "r5": {
+                    "100.64.0.0/32": "5555 5555 5555 5555 5555",
+                    "100.64.0.1/32": "5555 5555 5555 5555 5555 5555 5555",
+                    "100.64.0.2/32": "5555 5555 5555 1000 5555 2000 5555",
+                    "100.64.0.3/32": "5555 5555 5555 5555 1000 5555 2000 5555 5555",
+                    "100.64.0.4/32": "5555 5555 1000 2000 2000 3000",
+                },
+            },
+        },
+    }
+
+    # Simple lookup of remote peer ip by routers in session (local --> remote).
+    #
+    # Structure:
+    # peer_to_ip = {
+    #     local_rtr: {
+    #         peer_rtr: peer_ip
+    #     }
+    # }
+    peer_to_ip = {
+        "r1": {"r2": "10.2.10.2", "r5": "10.5.10.5"},
+        "r2": {"r1": "10.2.10.1", "r3": "10.2.30.3", "r4": "10.2.40.4"},
+        "r3": {"r2": "10.2.30.2", "r5": "10.5.30.5"},
+        "r4": {"r2": "10.2.40.2", "r5": "10.5.40.5"},
+        "r5": {"r1": "10.5.10.1", "r3": "10.5.30.3", "r4": "10.5.40.4"},
+    }
+
+    def __bgp_up(router):
+        """Return True if all configured peers are Established."""
+        output = json.loads(tgen.gears[router].vtysh_cmd("show ip bgp summary json"))
+        numPeers = output["ipv4Unicast"]["totalPeers"]
+        for attempts in range(0, 10):
```

Comment:
This seems error-prone. Can we use it here the same way as everywhere else? Example:

```
    def _bgp_converge():
        output = json.loads(
            tgen.gears["r1"].vtysh_cmd("show ip bgp summary json")
        )
        expected = {
            "10.0.0.2": {
                "bgpState": "Established",
            }
        }
        return topotest.json_cmp(output, expected)

    test_func = functools.partial(_bgp_converge)
    _, result = topotest.run_and_expect(test_func, None, count=60, wait=0.5)
    assert result is None, "Not all in Established state."
```